components/constellation/webview.rs

```diff
@@ -93,6 +101,38 @@ impl<WebView> WebViewManager<WebView> {
     pub fn unfocus(&mut self) {
         self.is_focused = false;
     }
+
+    pub fn set_webview_visibility(
+        &mut self,
+        top_level_browsing_context_id: TopLevelBrowsingContextId,
+        visible: bool,
+    ) {
+        debug_assert!(self.webviews.contains_key(&top_level_browsing_context_id));
+        if visible {
+            self.visible_webviews.insert(top_level_browsing_context_id);
+        } else {
+            self.visible_webviews.remove(&top_level_browsing_context_id);
+        }
+    }
+
+    pub fn set_native_window_visibility(&mut self, visible: bool) {
+        self.native_window_is_visible = visible;
+    }
```

Comment:
given that native window visibility was removed (<https://github.com/servo/servo/pull/30842/commits/9e307cfcefed26ecb65399ce12a7e2a7ef38e5d7>), i think we want something like

```rust
fn mark_webview_shown(webview_id: TopLevelBrowsingContextId);
fn mark_webview_not_shown(webview_id: TopLevelBrowsingContextId);
fn mark_webview_invisible(webview_id: TopLevelBrowsingContextId);
fn mark_webview_not_invisible(webview_id: TopLevelBrowsingContextId);
```

Reply:
Added in [`6d54829`](https://github.com/servo/servo/pull/31417/commits/6d5482974e6a17e98dd146626fe61a88aa582aee)